### PR TITLE
Add appmapping of some sub-projects

### LINKED ISSUE #1088: Enable mapping in (some) appmap-js test scripts

### PR DESCRIPTION
**Pull request by dividedmind**
Fixes #1088

### CODE REVIEW

**Comment by dividedmind** [comment]
To avoid useless overhead in CI, I changed it so that appmapping is not enabled by default, but available by running `yarn appmap`. @lachrist, could you take a look at the config files? I've mostly lifted them from your `dog` branch, but there could be differences.

File: packages/client/appmap.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+recorder: jest
+exclude:
+- \[anonymous\]
+packages:
+- regexp: (^|/)node_modules/
+  enabled: false
+- regexp: ^../
+  enabled: false
+- regexp: ^
+  enabled: true
```
Comment (lachrist) [inline packages/client/appmap.yml:12]: Note that `enabled: true` is the default if we want to make the config file shorter. I still prefer the explicit `enabled` value though.

File: packages/client/package.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "dist"
   ],
   "scripts": {
+    "appmap": "appmap-agent-js -- jest --no-cache",
```
Comment (lachrist) [inline packages/client/package.json:10]: In the next release, `--no-cache` will no longer be necessary.

File: packages/sequence-diagram/appmap.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+ordering: chronological
+command: yarn exec jest --no-cache
+recorder: jest
+exclude:
+- \[anonymous\]
+- matchAction
```
Comment (lachrist) [inline packages/sequence-diagram/appmap.yml:6]: Any reason why this is not scoped to a single package?

**Review by lachrist — COMMENTED** [review]
The config files looks legit. In  my dog branch I used a single config files to share some fields across different sub projects. But @kgilpin did not like it so one appmap config file per sub-project seems to be the way forward. I think this is okay to merge this if this is not part of CI yet. I still have some feedbacks to process.

**Comment by lachrist** [comment]
Now, pretty much all the sub projects can be appmaped with (various degree of success)

**Comment by brikelly** [comment]
@lachrist can we just close this PR now, since it sounds like other changes have addressed these issues in other PRs?

File: packages/client/appmap.yml (hunk 1 above)
Comment (dividedmind) [reply packages/client/appmap.yml:12]: Yeah, agreed.

File: packages/sequence-diagram/appmap.yml (hunk 3 above)
Comment (dividedmind) [reply packages/sequence-diagram/appmap.yml:6]: No reason, thanks.

**Review by kgilpin — APPROVED** [review]
Thank you for all this. Reminder - provide `name`, `language`, `appmap_dir` to *appmap.yml*.

**Pull request merged**